diff --git a/server/death_handler.cpp b/server/death_handler.cpp
index 07b25ca..9bda84c 100644
--- a/server/death_handler.cpp
+++ b/server/death_handler.cpp
@@ -605,6 +605,8 @@ void DeathHandler::SignalHandler (int sig, void * /* info */, void *secret)
   trace[1] = reinterpret_cast<void *> (uc->uc_mcontext.arm_pc);
 #elif defined(__aarch64__)     
   trace[1] = reinterpret_cast<void *> (uc->uc_mcontext.pc);
+#elif defined(__ppc__) || defined(__powerpc) || defined(__powerpc__) || defined(__POWERPC__)
+  trace[1] = reinterpret_cast<void *> (uc->uc_mcontext.regs->nip);
 #else
 #if !defined(__i386__) && !defined(__x86_64__)
 #error Only ARM, x86 and x86-64 are supported
